File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let idx = 0;
    
    while (idx < progresses.length) {
        let cnt = 0;
        
        for (let i = 0; i < progresses.length; i++) {
            progresses[i] += speeds[i];
        }
        
        while (idx < progresses.length && progresses[idx] >= 100) {
            cnt++;
            idx++;
        }
        
        if (cnt > 0) {
            answer.push(cnt);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    idx = 0
    
    while True:
        cnt = 0
        
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        
        for i in range(idx, len(progresses)):
            if progresses[i] >= 100:
                cnt += 1
            else:
                break
        
        if cnt > 0:
            answer.append(cnt)
            idx += cnt
        
        if len(progresses) == sum(answer):
            return answer